common-auth/src/main/java/com/hortonworks/registries/auth/server/KerberosBasicAuthenticationHandler.java

```diff
@@ -0,0 +1,215 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.hortonworks.registries.auth.server;
+
+import com.hortonworks.registries.auth.client.AuthenticationException;
+import com.hortonworks.registries.auth.util.KerberosName;
+import com.hortonworks.registries.auth.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.kerberos.authentication.KerberosAuthenticationProvider;
+import org.springframework.security.kerberos.authentication.sun.SunJaasKerberosClient;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Properties;
+
+/**
+ * The {@link KerberosBasicAuthenticationHandler} augments the Kerberos SPNEGO authentication mechanism with the Kerberos Basic Login authentication
+ * mechanism. If a user provides user credentials in a HTTPS, POST call, then a Kerberos login is attempted. In the authentication failure scenario,
+ * the SPNEGO sequence is invoked.
+ *
+ * <p>
+ * The supported configuration properties that are required in addition to those required by KerberosAuthenticationHandler are:
+ * <ul>
+ * <li>login.enabled: a boolean string to indicate whether the enabling of Kerberos login.</li>
+ * </ul>
+ */
+public class KerberosBasicAuthenticationHandler extends KerberosAuthenticationHandler {
```

Comment:
Can you talk about this authentication method in security.rst under docs pacakge ?